### ISSUE TITLE
Document NetSuite::Utilities module

### ISSUE THREAD

**Issue by monkbroc** [NetSweet/netsuite#360]
While rebasing a PR tonight I noticed changes in the NetSuite::Utilities module. There's a lot of good stuff in there! I was actually starting to think about doing an exponential backoff method for when NetSuite misbehaves.

Do you have examples of how to use the methods in there? I'll volunteer to write them into the README but I'd rather have a base to start from.

**Comment by iloveitaly** [comment]
@monkbroc Yeah, there's a bunch in there and a bunch I'd like to add. There's also a TON in `netsuite_rails` (https://github.com/netsweet/netsuite_rails). Unfortunately, I just don't have the time to put into documenting this stuff right now :(

Feel free to submit a PR!
